### ISSUE TITLE
CPP14.g4 only compatible with Java ANTLR

### ISSUE THREAD

**Issue by shaahmad** [antlr/grammars-v4#415]
@Camiloasc1 
As you know, ANTLR has targets for languages other than Java. I am using it in a Python script. But line 1093 i.e. the pure-specifier bit has a line of embedded Java:
`{if($val.text.compareTo("0")!=0) throw new InputMismatchException(this, $val.text);}`

This causes Python to not be able to interpret the ANTLR output, raising a Syntax Error.

It would be nice to either indicate this in the "Possible Issues" section or somehow work around this.

**Comment by Camiloasc1** [comment]
Yes, it's a very tricky issue to solve without embedded code (considering the '0' as a token breaks the lexer), but I guess you can also verify this rule in your listeners or visitors.
I'm not sure if `$val` works in python or has another name, but if does you could try with:
```
if not $val.text == "0":
    raise InputMismatchException(this);
```

Check #416.

**Comment by Hyunsu-Lee** [comment]
@Camiloasc1 I think that your code need to change to like this:
`purespecifier 
: 
    Assign val = Octalliteral 
    {if not $val.text == '0': 
        raise InputMismatchException(this)} 
; 
`

**Comment by Marti2203** [comment]
Is this fixed? @Camiloasc1 @Hyunsu-Lee

**Comment by Hyunsu-Lee** [comment]
> Is this fixed? @Camiloasc1 @Hyunsu-Lee

I was using changed code separately.
Today, I applied this change to the fork and made a pull request, but all checks failed.

**Comment by Marti2203** [comment]
I can check out the changes and help with the PR then
